client/src/components/AddCard.js

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from 'react'
+import { makeStyles } from '@material-ui/core/styles'
+import {
+    Button,
+    Typography,
+    Snackbar,
+} from '@material-ui/core'
+import Alert from '@material-ui/lab/Alert'
+import { loadStripe } from '@stripe/stripe-js'
+import {
+    CardNumberElement,
+    CardExpiryElement,
+    CardCvcElement,
+    Elements,
+    useStripe,
+    useElements,
+} from '@stripe/react-stripe-js'
+import { useAuth } from "./UserContext";
+const stripeKey = require('../config/default.json').stripeKey
+
+function CheckoutForm() {
+    const stripe = useStripe()
+    const elements = useElements()
+    const classes = useStyles()
+
+    const [openAlert, setOpenAlert] = useState(false)
+    const [severity, setSeverity] = useState("success") //success, error, info
+    const [alertMessage, setAlertMessage] = useState("")
+
+    const { authTokens, setAuthTokens } = useAuth();
+    const [user] = useState(authTokens ? authTokens.user : null);
+
+    const handleSubmit = async (event) => {
+        event.preventDefault()
+        const { error, paymentMethod } = await stripe.createPaymentMethod({
+            type: 'card',
+            card: elements.getElement(CardNumberElement)
+        })
+        if (error) {
+            handleAlert(error.message, "error")
+
+        }
+        if (paymentMethod) {
+            console.log(paymentMethod)
+            const _user = user
+            _user.hasPaymentInfoSaved = true
+            _user.paymentInfo = { cardType: paymentMethod.card.brand, last4: paymentMethod.card.last4 }
+            _user.paymentId = paymentMethod.id
```

Comment:
```suggestion
const _user = {
  ...user,
  hasPaymentInfoSaved: true,
  paymentInfo: { cardType: paymentMethod.card.brand, last4: paymentMethod.card.last4 },
  paymentId: paymentMethod.id
}
```

Reply:
suggestion to reduce the amount of time you have to type _user :)

Reply:
Yes that looks cleaner, thanks.